### DISCUSSION TITLE
canvasToTexture crash

### DISCUSSION THREAD

**Discussion by wirmachenbunt** [cables-gl/cables_docs#730] [Answers & Questions]
<img width="1475" alt="image" src="https://github.com/cables-gl/cables_docs/assets/1138673/30f0e876-ef31-4fb7-aef6-0c4c2ebd294e">
connecting a div element to canvasToTexture results in a crash
is this op broken or user error ?

**Comment by pandrr** [comment]
thats not possible its CANVAS to texture, it is not possible to convert a div to a texture

**Reply by wirmachenbunt** [reply]
that's why i wrote "user error ?", i wasn't sure if there is some magic going on e.g. the div is internally rendered to a canvas and then converted to a texture.

**Reply by pandrr** [reply]
i don't know a way to detect if an element can used in texImage2d. i dont have a classification for user-errors or not-user-errors. what do you mean where is magic going on ?

**Comment by pandrr** [comment]
from texImage2D docs:

The following types can always be used as a pixel source for the texture:

[ImageData](https://developer.mozilla.org/en-US/docs/Web/API/ImageData),
[HTMLImageElement](https://developer.mozilla.org/en-US/docs/Web/API/HTMLImageElement),
[HTMLCanvasElement](https://developer.mozilla.org/en-US/docs/Web/API/HTMLCanvasElement),
[HTMLVideoElement](https://developer.mozilla.org/en-US/docs/Web/API/HTMLVideoElement),
[ImageBitmap](https://developer.mozilla.org/en-US/docs/Web/API/ImageBitmap).


https://developer.mozilla.org/en-US/docs/Web/API/WebGLRenderingContext/texImage2D
